Makefile

```diff
@@ -10,9 +10,13 @@ ARCH := $(if $(CARGO_BUILD_TARGET),$(firstword $(subst -, ,$(CARGO_BUILD_TARGET)
 RELEASE_VERSION = $(shell tools/localversion)
 RELEASE_NAME ?= $(shell $(CARGO) metadata --no-deps --format-version=1 | jq -r '.packages | .[] | select(.name=="retis") | .metadata.misc.release_name')
 
+DEFAULT_FEATURES = python
+FEATURES ?= $(DEFAULT_FEATURES)
```

Comment:
> Is there a reason to allow changing the set of features? We could always build it when building the Retis binary, since the Python library has its own target.
> 

You mean if users want to build with some other feature, they should call cargo manually?

> With that I'm wondering if the `python` feature is useful, we could do the same by checking the `python-lib` feature is _not_ selected. This also has the advantage of really excluding one for the other.

`python-lib` feature is for `retis-events` only. We need a global (used on the binary) that will enable the python features on subcrates and build the python cli command.

Reply:
> > Is there a reason to allow changing the set of features? We could always build it when building the Retis binary, since the Python library has its own target.
> 
> You mean if users want to build with some other feature, they should call cargo manually?

I'm wondering if there's a use case letting the user select features when building Retis. I don't think we plan on letting Retis be built with different sets of features; with this PR the only possible one would be w/o the `python` one but I don't think we should allow that. When we do need different features enabled, we can use targets (that's the current logic with the Python lib).

> > With that I'm wondering if the `python` feature is useful, we could do the same by checking the `python-lib` feature is _not_ selected. This also has the advantage of really excluding one for the other.
> 
> `python-lib` feature is for `retis-events` only. We need a global (used on the binary) that will enable the python features on subcrates and build the python cli command.

I was thinking building `retis-events` w/o selecting a feature should include Python support (not the lib) and then have a `phyton-lib` feature if we do want to build the Python library. The `phyton-lib` can then include additional things needed and exclude ones that are incompatible (currently under the `python-embed` feature). I would find this simpler and easier to maintain, also IMO it's logical: by default it can be used as a Rust crate and included somewhere else, but has a special mode to build a standalone Python library.

Or maybe the goal is to let people use `retis-events` w/o getting `pyo3` at all if needed? We can think about this when there are additional users and this is requested maybe.

I might be missing some dots though.

Reply:
> > > Is there a reason to allow changing the set of features? We could always build it when building the Retis binary, since the Python library has its own target.
> > 
> > 
> > You mean if users want to build with some other feature, they should call cargo manually?
> 
> I'm wondering if there's a use case letting the user select features when building Retis. I don't think we plan on letting Retis be built with different sets of features; with this PR the only possible one would be w/o the `python` one but I don't think we should allow that. When we do need different features enabled, we can use targets (that's the current logic with the Python lib).
> 
> > > With that I'm wondering if the `python` feature is useful, we could do the same by checking the `python-lib` feature is _not_ selected. This also has the advantage of really excluding one for the other.
> > 
> > 
> > `python-lib` feature is for `retis-events` only. We need a global (used on the binary) that will enable the python features on subcrates and build the python cli command.
> 
> I was thinking building `retis-events` w/o selecting a feature should include Python support (not the lib) and then have a `phyton-lib` feature if we do want to build the Python library. The `phyton-lib` can then include additional things needed and exclude ones that are incompatible (currently under the `python-embed` feature). I would find this simpler and easier to maintain, also IMO it's logical: by default it can be used as a Rust crate and included somewhere else, but has a special mode to build a standalone Python library.
> 

AFAIK, features cannot be disabled, only enabled. So `python-lib` cannot disable features from `python-embed` if those are now enabled by default.

> Or maybe the goal is to let people use `retis-events` w/o getting `pyo3` at all if needed? We can think about this when there are additional users and this is requested maybe.
> 

Yes, that was the original rationale. That and building `retis` without pyo3 which reduces the size of the binary by 14M.


> I might be missing some dots though.

Reply:
I guess if we want to push users a bit more towards python we can make `python` a default feature and still allow people to build with "--no-default-features" if desired.